### PR TITLE
Add write support for entries larger 8 GB

### PR DESCRIPTION
**Pull request by ususdei**
Most programs support an extension for tar which allow to write large integers in binary coded format denoted by a set first bit.

### CODE REVIEW

**Comment by piranna** [comment]
Do you have some references for that?

**Comment by ususdei** [comment]
Sure:

There reference is here: http://www.gnu.org/software/tar/manual/html_node/Extensions.html#Extensions

Additionally `tar-stream` can already read this format in `decodeOct()`.
It's also what `node-tar` does in https://github.com/npm/node-tar/blob/main/lib/large-numbers.js

**Review by piranna — APPROVED** [review]

File: headers.js [hunk 1]
```diff
@@ -92,12 +92,29 @@ var cksum = function (block) {
   return sum
 }
 
+var encodeSizeBin = function(num, buf, off) {
+    buf[off] = 0x80;
+    for (var i=11; i>0; i--) {
+        buf[off+i] = num & 0xff
+        num = Math.floor(num / 0x100)
+    }
+}
```
Comment (mafintosh) [inline headers.js:101]: two space indentation

File: headers.js [hunk 2]
```diff
@@ -92,12 +92,29 @@ var cksum = function (block) {
   return sum
 }
 
+var encodeSizeBin = function(num, buf, off) {
```
Comment (mafintosh) [inline headers.js:95]: space after function, `function (`

**Comment by mafintosh** [comment]
There are some lint errors. I commented on some, Standard should tell you the rest (npm run test)

**Comment by ususdei** [comment]
Updated. Thanks!

**Comment by ususdei** [comment]
@mafintosh Any update on this? Did I miss something? I would love to see this upstreamed.

**Comment by jeff-al** [comment]
Hi @mafintosh, is this solution correct? I think it is a very important change that many of us need

File: headers.js [hunk 3]
```diff
@@ -92,12 +92,29 @@ var cksum = function (block) {
...
+
+
 var encodeOct = function (val, n) {
   val = val.toString(8)
   if (val.length > n) return SEVENS.slice(0, n) + ' '
   else return ZEROS.slice(0, n - val.length) + val + ' '
 }
 
+var encodeSize = function (num, buf, off) {
+    if (num.toString(8).length > 11)
```
[the remark this answers is not in the document]
Comment (gitadvisor) [reply headers.js]: > Use curly brackets, ie `if (...) {` and two space indent

**Comment by mafintosh** [comment]
landing in 3.1.0

**Pull request closed without merging**